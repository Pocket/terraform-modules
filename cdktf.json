{
  "language": "typescript",
  "app": "npm run --silent compile && node dist/src/example.js",
  "terraformProviders": [
    "pagerduty/pagerduty@~> 1.8.0",
    "archive@~> 1.3.0",
    "local@~> 2.1.0"
  ]
}
