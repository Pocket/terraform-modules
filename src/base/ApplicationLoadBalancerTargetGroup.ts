import { Resource } from 'cdktf';
import { Construct } from 'constructs';
import { AlbTargetGroup } from '../../.gen/providers/aws';

export interface ApplicationLoadBalancerTargetGroupProps {
  interval: number;
  path: string;
  protocol: string;
  protocolHealthCheck: string;
  healthyThreshold: number;
  unhealthyThreshold: number;
  timeout: number;
  tags?: { [key: string]: string };
  vpcId: string;
  port: number;
  name: string;
  targetType: string;
  deregistrationDelay: number;
  dependsOn?: Resource[];
}

/**
 * Generates an ALB Target Group
 */
export class ApplicationLoadBalancerTargetGroup extends Resource {
  public readonly albTargetGroup: AlbTargetGroup;
  constructor(
    scope: Construct,
    name: string,
    config: ApplicationLoadBalancerTargetGroupProps
  ) {
    super(scope, name);

    const albTargetGroupConfig = {
      healthCheck: [
        {
          interval: config.interval,
          path: config.path,
          protocol: config.protocol,
          healthyThreshold: config.healthyThreshold,
          unhealthyThreshold: config.unhealthyThreshold,
          timeout: config.timeout,
        },
      ],
      vpcId: config.vpcId,
      protocol: config.protocolHealthCheck,
      port: config.port,
      name: config.name,
      targetType: config.targetType,
      deregistrationDelay: config.deregistrationDelay,
      tags: config.tags,
    };
    if (config.dependsOn) {
      albTargetGroupConfig[`dependsOn`] = config.dependsOn;
    }
    this.albTargetGroup = new AlbTargetGroup(
      this,
      `alb_target_group`,
      albTargetGroupConfig
    );
  }
}
