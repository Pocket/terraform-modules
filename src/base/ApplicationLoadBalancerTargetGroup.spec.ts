import { Testing, TerraformStack } from 'cdktf';
import { ApplicationLoadBalancerTargetGroup } from './ApplicationLoadBalancerTargetGroup';

test('renders target group without tags', () => {
  const app = Testing.app();
  const stack = new TerraformStack(app, 'test');

  new ApplicationLoadBalancerTargetGroup(stack, 'testTargetGroup', {
    interval: 5,
    path: '/',
    protocol: 'HTTP',
    protocolHealthCheck: 'HTTP',
    healthyThreshold: 2,
    unhealthyThreshold: 3,
    timeout: 5,
    vpcId: '123',
    port: 81,
    name: 'test',
    targetType: 'ip',
    deregistrationDelay: 5,
  });

  expect(Testing.synth(stack)).toMatchSnapshot();
});

// test('renders an ALB Target with tags', () => {
//   const app = Testing.app();
//   const stack = new TerraformStack(app, 'test');
//
//   new ApplicationLoadBalancer(stack, 'testALB', {
//     prefix: 'test-',
//     alb6CharacterPrefix: 'TEST',
//     vpcId: '123',
//     subnetIds: ['a', 'b'],
//     tags: {
//       name: 'thedude',
//       hobby: 'bowling',
//     },
//   });

// expect(Testing.synth(stack)).toMatchSnapshot();
// });
