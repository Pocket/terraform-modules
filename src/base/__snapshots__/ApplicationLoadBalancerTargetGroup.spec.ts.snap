// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders target group without tags 1`] = `
"{
  \\"//\\": {
    \\"metadata\\": {
      \\"version\\": \\"stubbed\\",
      \\"stackName\\": \\"test\\"
    }
  },
  \\"resource\\": {
    \\"aws_alb_target_group\\": {
      \\"test_testTargetGroup_albtargetgroup_E58AF39E\\": {
        \\"deregistration_delay\\": 5,
        \\"name\\": \\"test\\",
        \\"port\\": 81,
        \\"protocol\\": \\"HTTP\\",
        \\"target_type\\": \\"ip\\",
        \\"vpc_id\\": \\"123\\",
        \\"health_check\\": [
          {
            \\"interval\\": 5,
            \\"path\\": \\"/\\",
            \\"protocol\\": \\"HTTP\\",
            \\"healthy_threshold\\": 2,
            \\"unhealthy_threshold\\": 3,
            \\"timeout\\": 5
          }
        ],
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/testTargetGroup/alb_target_group\\",
            \\"uniqueId\\": \\"test_testTargetGroup_albtargetgroup_E58AF39E\\"
          }
        }
      }
    }
  }
}"
`;
